docs/api/config.md

```diff
@@ -495,7 +558,17 @@ Each pattern can start with a variable that resolves to a system base directory.
 Can be any of the following types:
 
 - `string[]`: A list of paths that are allowed by this scope.
-- `object`: A complete scope configuration.
+- `{ "allow": [string], "deny": [string], "requireLiteralLeadingDot": boolean? }`: A complete scope configuration.
```

Comment:
do we want to embed definition here instead of just `object`? because it is epxlained belowe

Reply:
There's some enum variants with multiple objects so it feels weird to have several `- object:` lines IMO. But longer definitions are indeed weird.

Reply:
What do you think @amrbashir @lorenzolewis